src/functions/sendCandidateResults/application/service/__tests__/fault-provider.spec.ts

```diff
@@ -356,4 +357,77 @@ describe('fault-provider', () => {
 
     });
   });
+
+  describe('getVehicleCheckFaultCount', () => {
+    it('should return 0 when no vehicle checks are present', () => {
+      const data: CatFUniqueTypes.VehicleChecks = {};
+      const count: number = getVehicleCheckFaultCount(data, CompetencyOutcome.DF);
+      expect(count).toEqual(0);
+    });
+
+    it('should return a count of 2 as there are two DF present in the showMeQuestions', () => {
+      const data: CatFUniqueTypes.VehicleChecks = {
+        showMeQuestions: [
+          {
+            outcome: 'DF',
+          },
+          {
+            outcome: 'P',
+          },
+          {
+            outcome: 'DF',
+          },
+        ],
+      };
+      const count: number = getVehicleCheckFaultCount(data, CompetencyOutcome.DF);
+      expect(count).toEqual(2);
+    });
+
+    it('should return a count of 2 as there are two DF present in the tellMeQuestions', () => {
+      const data: CatFUniqueTypes.VehicleChecks = {
+        tellMeQuestions: [
+          {
+            outcome: 'DF',
+          },
+          {
+            outcome: 'P',
+          },
+          {
+            outcome: 'DF',
+          },
+        ],
+      };
+      const count: number = getVehicleCheckFaultCount(data, CompetencyOutcome.DF);
+      expect(count).toEqual(2);
+    });
+
+    it('should return a count of 4 as there are two DF present in the showMeQuestions & tellMeQuestions', () => {
+      const data: CatFUniqueTypes.VehicleChecks = {
+        showMeQuestions: [
+          {
+            outcome: 'DF',
+          },
+          {
+            outcome: 'P',
+          },
+          {
+            outcome: 'DF',
+          },
+        ],
+        tellMeQuestions: [
+          {
+            outcome: 'DF',
+          },
+          {
+            outcome: 'P',
+          },
+          {
+            outcome: 'DF',
+          },
+        ],
+      };
+      const count: number = getVehicleCheckFaultCount(data, CompetencyOutcome.DF);
+      expect(count).toEqual(4);
+    });
+  });
```

Comment:
Confused by the tests - for home tests there is only 1 show me 1 tell me, and even if you have 2 wrong results it only counts as 1  df.

Reply:
This was the duplicated function I mentioned, its just doing a count of the number of specified outcomes in the show me/tell me questions array.